functions/src/compute-statistics.ts

```diff
@@ -24,25 +30,38 @@ export const computeStatistics = functions.https.onRequest(
       return;
     }
 
-    await firebaseTools.firestore.delete("stats", {
-      project: config.service_account.project_id,
-      recursive: true,
-      yes: true
-    });
-
     const votes = await db.collection("vote").get();
-    for (const vote of votes.docs) {
-      const voteData = vote.data() as Vote;
-      await updateStats(
-        voteData.value,
-        vote.id,
-        firebase
-          .firestore()
-          .collection(`stats-campaign`)
-          .doc(vote.id),
-        { agency: voteData.agency }
-      );
-    }
-    res.send("done");
+
+    votes.docs
+      .map(vote => {
+        const voteData = vote.data() as Vote;
+        return getStatsRefsToUpdate(voteData).map(row => {
+          return {
+            ref: row.ref,
+            voteValue: voteData.value,
+            additionalFields: row.additionnalFields
+          };
+        });
+      })
+      .reduce((accumulator, row) => [...accumulator, ...row], [])
+      .reduce((countersByRef, row) => {
+        const counters = countersByRef.get(row.ref.path);
```

Comment:
I see you use string paths as keys. Refs as Map keys don't work?

Reply:
No, every campaign showed with 1 vote

Reply:
OK so let's switch back to an object then. Using a Map is surprising to the reader so it has to be justified.